pppd/main.c

```diff
@@ -747,6 +750,114 @@ set_ifunit(int iskey)
     }
 }
 
+/*
+ * net-* scripts to be run come through here.
+ */
+static void run_net_script(char* script, int wait)
+{
+    char strspeed[32];
+    char *argv[6];
+
+    slprintf(strspeed, sizeof(strspeed), "%d", baud_rate);
+
+    argv[0] = script;
+    argv[1] = ifname;
+    argv[2] = devnam;
+    argv[3] = strspeed;
+    argv[4] = ipparam;
+    argv[5] = NULL;
+
+    run_program(script, argv, 0, NULL, NULL, wait);
+}
+
+/*
+ * set_ifup - called in order to set the ppp interface to up, if not already
+ * brought up.
+ *
+ * It's important that the sub-protocol not be a dynamically allocated string, ie,
+ * use set_ipup("IPCP") ...
+ */
+int
+set_ifup(const char* name)
+{
+    int i;
+    const char** t;
+
+    if (!up_protos) {
+	char iftmpname[IFNAMSIZ];
+	int ifindex = if_nametoindex(ifname);
+	run_net_script(PPP_PATH_NET_PREUP, 1);
+	if (if_indextoname(ifindex, iftmpname) && strcmp(iftmpname, ifname)) {
+	    info("Detected interface name change from %s to %s.", ifname, iftmpname);
+	    strcpy(ifname, iftmpname);
```

Comment:
Need to call script_setenv here for IFNAME?  What do I pass for iskey?

Only case where set_ifunit is called with iskey=0 is under multilink, in which case pppd joining a bundle I suspect (but don't KNOW) that the bundle interface will already be UP and thus set_ifup will never be called.

We've only ever used multilink in a handful of cases, and those use-cases are long-gone now so don't have an environment where I can scope and check this.